code/obj/item/grenades.dm

```diff
@@ -893,45 +893,92 @@ PIPE BOMBS + CONSTRUCTION
 	stamina_damage = 5
 	stamina_cost = 5
 	stamina_crit_chance = 5
+	var/slashed = 0 // has it been emptied out? if so, better dud!
+	var/primer_burnt = 0 // avoid priming a firework multiple times, that doesn't make sense!
+	var/primed = 0 // cutting open lit fireworks is a BAD idea
+
+	New()
+		..()
+		create_reagents(10)
+		reagents.add_reagent("magnesium", 10)
 
 	afterattack(atom/target as mob|obj|turf|area, mob/user as mob)
 		if (user.equipped() == src)
-			if (user.bioHolder.HasEffect("clumsy"))
+			if (src.primer_burnt)
+				boutput(user, "<span class='alert'>You can't light a firework more than once!</span>")
+				return
+			else if (src.slashed)
+				boutput(user, "<span class='alert'>You prime the firework! [det_time/10] seconds!</span>")
+				SPAWN_DBG( src.det_time )
+					boutput(user, "<span class='alert'>The firework probably should have exploded by now. Fuck.</span>")
+					src.primer_burnt = 1
+					return
+			else if (user.bioHolder.HasEffect("clumsy"))
 				boutput(user, "<span class='alert'>Huh? How does this thing work?!</span>")
+				src.primed = 1
 				SPAWN_DBG( 5 )
 					boom()
 					return
 			else
 				boutput(user, "<span class='alert'>You prime the firework! [det_time/10] seconds!</span>")
+				src.primed = 1
 				SPAWN_DBG( src.det_time )
 					boom()
 					return
 
 	proc/boom()
 		var/turf/location = get_turf(src.loc)
 		if(location)
+			src.visible_message("<span class='alert'>\The [src] explodes!</span>") // moving to avoid race condition
 			if(prob(10))
 				explosion(src, location, 0, 0, 1, 1)
+				qdel (src) // moving to avoid race condition
```

Comment:
reasoning behind the move? there's no race condition

also extremely gross spacing

Reply:
i was concerned over calling qdel twice on the same src after cutting open a primed firework. After a discussion in the discord and some tests I see why this is unnecessary.

also tried to pretty it up for readability